### ISSUE TITLE
Two players blocking each other from playing in the same game 

### ISSUE THREAD

**Issue by ghost** [freeciv/freeciv-web#145]
This is the third time that I've encountered the bug, but this time it happened to me. It seems to be some kind of conflict between two random players and the second player won't be able to join a multi game unless the 1st one leaves, after which they can't join again. You get a blank freeciv web screen. The problem persists (possibly for the duration of the game?). This time it happened between me and Lucas Hudson. 
I have a VPN and I tried switching my IP to different countries, but that didn't allow me to join.
My OS - Bodhi Linux (based on Ubuntu), using Mozilla Firefox
Lucas' - Windows 10 - Google Chrome
I'm attaching screenshots of the blank screen with the javascript console open. I tried joining both as Rhue and some random gibberish, with the same results.
Then Lucas suddenly left and I was able to get in.
![rhue](https://user-images.githubusercontent.com/30433768/34650071-c0502802-f3ba-11e7-8ff1-e8fe29aa3cf5.png)
![gibberish](https://user-images.githubusercontent.com/30433768/34650073-c69a3a86-f3ba-11e7-8928-297782ee39b7.png)
![lucas leaves](https://user-images.githubusercontent.com/30433768/34650074-c7b84a5c-f3ba-11e7-8503-921f17ea2a36.png)

**Comment by ghost** [comment]
Oh, I tried joining through another browser, Midori, and that failed as well.

**Comment by ghost** [comment]
Can you please report the exact steps required to reproduce this problem?
Do single player games work?

**Comment by ghost** [comment]
Yes, single player games work as this is caused by some conflict between two players in multiplayer games. I sadly can't tell you the exact steps, because of the nature of the bug. It's the third time it's happened, and between different players each time. The only way you could reproduce it is to play multiplayer enough to see it happen ;), but sadly because there aren't that many multi games at this moment I guess people don't encounter the bug that much.

**Comment by ghost** [comment]
Nor do the games that are played have many players, so even harder to stumble across the bug.

**Comment by ghost** [comment]
Thanks for the information provided so far.

Can you please try starting 10 multiplayer games, and report how often this bug happens?

**Comment by ghost** [comment]
I would have to have volunteers willing to test this with me. 
It's not easy to find players usually.

**Comment by ghost** [comment]
Oh, by the way, the problem has only occurred in games with 4 or more players, I believe. Which is another reason why it hadn't been reported before, because multi games in freeciv web usually have only 2 players.

**Comment by lonemadmax** [comment]
May be CONNECTION_LIMIT in freeciv-proxy? It would be consequent with working after a user disconnects and then that user being the one who cannot enter. The only message out of place in the screenshots is just the notification of the websocket connection being closed.

@RhueOfL: Is the last screenshot from the same game? The players list on the right don't look like a "clean" multiplayer game just after opening one and four players joining. What had been changed? Or it may not matter, if you have found the same problem without touching anything.

**Comment by ghost** [comment]
CONNECTION_LIMIT is 1000, though.

**Comment by ghost** [comment]
Uh, that's a game I tried to hotjoin. It's an earth scenario, it has specific ais selected, those players you can see are AI players with AI names. Down the list you could see one more human players and 2 were playing with AI names, because they were hotjoiners, too. Oh by the way, I remember the bug already happening once between Lucas and somebody else, so maybe there's some connection to Google Chrome/Windows 10? The other time it was GusKurtis and DarkDusk, but I don't know if either of them is using Windows10/Chrome.

And yes, the bug would appear both on creating a new game and hotjoining.
